src/openrct2-ui/windows/News.cpp

```diff
@@ -204,41 +184,38 @@ static void window_news_scrollgetsize(rct_window* w, int32_t scrollIndex, int32_
 static void window_news_scrollmousedown(rct_window* w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords)
 {
     int32_t itemHeight = window_news_get_item_height();
-    int32_t i, buttonIndex;
 
-    buttonIndex = 0;
+    int32_t i = 0;
+    int32_t buttonIndex = 0;
     auto mutableScreenCoords = screenCoords;
-    for (i = NEWS_ITEM_HISTORY_START; i < MAX_NEWS_ITEMS; i++)
+    for (const auto& newsItem : gNewsItems.GetArchived())
     {
-        if (news_item_is_empty(i))
-            break;
-
         if (mutableScreenCoords.y < itemHeight)
         {
-            NewsItem* const newsItem = news_item_get(i);
-            if (newsItem->Flags & NEWS_FLAG_HAS_BUTTON || mutableScreenCoords.y < 14 || mutableScreenCoords.y >= 38
+            if (newsItem.Flags & NEWS_FLAG_HAS_BUTTON || mutableScreenCoords.y < 14 || mutableScreenCoords.y >= 38
                 || mutableScreenCoords.x < 328)
             {
                 buttonIndex = 0;
                 break;
             }
-            else if (mutableScreenCoords.x < 351 && news_type_properties[newsItem->Type] & NEWS_TYPE_HAS_SUBJECT)
+            else if (mutableScreenCoords.x < 351 && news_type_properties[newsItem.Type] & NEWS_TYPE_HAS_SUBJECT)
             {
                 buttonIndex = 1;
                 break;
             }
-            else if (mutableScreenCoords.x < 376 && news_type_properties[newsItem->Type] & NEWS_TYPE_HAS_LOCATION)
+            else if (mutableScreenCoords.x < 376 && news_type_properties[newsItem.Type] & NEWS_TYPE_HAS_LOCATION)
             {
                 buttonIndex = 2;
                 break;
             }
         }
         mutableScreenCoords.y -= itemHeight;
+        ++i;
```

Comment:
```suggestion
        i++;
```

For consistency with the rest of the code.